scarlet/display.py

```diff
@@ -364,79 +394,106 @@ def show_sources(
             observation is not None
         ), "Provide matched observation to show observed frame"
 
-    panels = sum((show_model, show_observed, show_rendered, show_sed))
+    panels = sum((show_model, show_observed, show_rendered, show_spectrum))
     if figsize is None:
         figsize = (3 * panels, 3 * len(list(sources)))
-    fig, ax = plt.subplots(len(list(sources)), panels, figsize=figsize)
+    fig, ax = plt.subplots(len(list(sources)), panels, figsize=figsize, squeeze=False)
+
+    marker_kwargs = {"mew": 1, "ms": 10}
+    box_kwargs = {"facecolor": "none", "edgecolor": "w", "lw": 0.5}
+
     for k, src in enumerate(sources):
-        if hasattr(src, "center"):
+
+        model_frame = src.model_frame
+
+        if hasattr(src, "center") and src.center is not None:
             center = np.array(src.center)[::-1]
         else:
             center = None
 
-        panel = 0
+        if add_boxes:
+            start, stop = src.bbox.start[-2:][::-1], src.bbox.stop[-2:][::-1]
+            points = (start, (start[0], stop[1]), stop, (stop[0], start[1]))
+            box_coords = [
+                observation.frame.get_pixel(model_frame.get_sky_coord(p))
+                for p in points
+            ]
 
+        # model in its bbox
+        panel = 0
         model = src.get_model()
-        if isinstance(src, ComponentTree):
-            seds = []
-            for component in src:
-                model_ = component.get_model()
-                seds.append(model_.sum(axis=(1, 2)))
-        else:
-            seds = [model.sum(axis=(1, 2))]
 
-        extent = get_extent(src.bbox)
-        rendered_box = src.bbox
+        # sed needs to be evaluated in the source box to prevent truncation
+        if show_spectrum:
+            if isinstance(src, ComponentTree):
+                spectra = []
+                for component in src:
+                    model_ = component.get_model()
+                    spectra.append(model_.sum(axis=(1, 2)))
+            else:
+                spectra = [model.sum(axis=(1, 2))]
```

Comment:
For (no good) reasons, we overwrite `model` with its rendered version, from which I cannot simply read of the photometry because it's in a different frame. It might be truncated at the edge of the observation.

I'll refactor to make this more transparent.